src/Feldspar/Onnx/onnxToFeld.hs

```diff
@@ -194,39 +199,44 @@ showElemT TD.STRING     = "String"
 showElemT TD.BOOL       = "Bool"
 showElemT TD.UNDEFINED  = error $ "onnxToFeld.showElemT: UNDEFINED not implemented"
 
-showVI :: V.ValueInfoProto -> String
-showVI v = nStr ++ " : " ++ tyStr
-  where nStr = toStr $ fromJust $ V.name v
+showVI :: V.ValueInfoProto -> L.ByteString
+showVI v = nStr <> " : " <> tyStr
+  where nStr = strFromJ $ V.name v
         tyStr = showType $ fromJust $ T.value $ fromJust $ V.type' v
 
-showType :: TV.Value -> String
+showType :: TV.Value -> L.ByteString
 showType TV.Tensor_type{TV.tensor_type = t}
   = showTensorType (TT.elem_type t)
                    (map TSP.value $ D.toList $ TSP.dim $ fromJust $ TT.shape t)
 showType TV.Sequence_type{} = "Sequence"
 showType TV.Map_type{} = "Map"
 
-showDim :: Maybe TSP.Value -> String
+showDim :: Maybe TSP.Value -> L.ByteString
 showDim (Just (TSP.Dim_value i)) = show i
-showDim (Just (TSP.Dim_param p)) = toStr p
+showDim (Just (TSP.Dim_param p)) = unUtf8 p
 showDim Nothing = "*"
 
-toStr :: H.Utf8 -> String
-toStr (H.Utf8 s) = U.toString s
+-- | Unpack a ByteString from the Utf8 newtype
+unUtf8 :: H.Utf8 -> L.ByteString
```

Comment:
Isn't this a cut and paste of `utf8` from Protobufs?

Reply:
Of course, but I looked in the wrong place in the documentation and did not find it. Will change.